realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -927,6 +927,11 @@ private void emitToStringMethod(JavaWriter writer) throws IOException {
         writer.emitEmptyLine();
     }
 
+    /**
+     * Currently, the hash value emitted from this could suddenly change as an object's index might
+     * alternate along with the changes of the Realm. It is therefore advisable not to solely
```

Comment:
The internal Javadoc don't need to document the public behaviour IMO, but more describe what is actually happening.
Something like "Currently, the hash value emitted from this could suddenly change as an object's index might change due to Realm Java using `move_last_over`. Hash codes should therefor not be considered stable, i.e. don't save them in a HashSet or use them as a key in a HashMap.


alternate -> " change due to other changes in the Realm